src/main/java/nextstep/qna/domain/Answers.java

```diff
@@ -0,0 +1,50 @@
+package nextstep.qna.domain;
+
+import nextstep.qna.CannotDeleteException;
+import nextstep.users.domain.NsUser;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class Answers {
+    private final List<Answer> answers;
+
+    public Answers() {
+        this.answers = new ArrayList<>();
+    }
+
+    public Answers(List<Answer> answerList) {
+        this.answers = new ArrayList<>(answerList);
+    }
+
+    public void add(Answer answer) {
+        answers.add(answer);
+    }
+
+    public List<Answer> getAnswers() {
+        return Collections.unmodifiableList(answers);
+    }
+
+    public void delete(NsUser writer) throws CannotDeleteException {
+        this.delete(writer, null);
+    }
+
+    public void delete(NsUser writer, DeleteHistories deleteHistories) throws CannotDeleteException {
+        try {
+            for (Answer answer : answers) {
+                answer.delete(writer);
+                if(deleteHistories != null) {
+                    deleteHistories.add(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now());
+                }
+            }
+        } catch (CannotDeleteException e) {
+            throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
+        }
+    }
+
+    public boolean isDeleted(){
+        return answers.stream().allMatch(answer->answer.isDeleted());
+    }
```

Comment:
```suggestion
    public void delete(NsUser writer) throws CannotDeleteException {
        try {
            for (Answer answer : answers) {
                answer.delete(writer);
            }
        } catch (CannotDeleteException e) {
            throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
        }
    }

    public void delete(NsUser writer, DeleteHistories deleteHistories) throws CannotDeleteException {
        this.delete(writer);

        if (deleteHistories != null) {
            deleteHistories.addAllAnswers(answers);
        }
    }

    public boolean isDeleted() {
        return answers.stream().allMatch(Answer::isDeleted);
    }
```
- 테스트만을 위해 운영 코드를 사용하기 보단, 가급적 같이 활용해주시는 게 좋습니다.
- 이력 저장하는 로직 역시 일급컬렉션에 역할을 위임할 수 있어요.
- 메서드를 기능 단위로 쪼개면 좀 더 테스트하기 용이해질 수 있어요.